src/lazy/encoder/text/value_writer.rs

```diff
@@ -114,6 +104,7 @@ struct TextContainerWriter_1_0<'a, W: Write> {
     // The Ion type of the container using this TextContainerWriter_1_0. This value is only
     // used for more informative error messages.
     ion_type: IonType,
+    value_delimiter: &'static str,
```

Comment:
🗺️ When `StuctWriter`s go to serialize a field, they use `FieldEncoder::encode_field_name(...)` and a `ValueWriter` to encode it. In text, this means that the `ValueWriter` is responsible for writing its trailing delimiter where applicable.

I considered adding a `write_delimiter` method to `StructWriter`/`FieldEncoder`, but didn't like that it wouldn't be applicable to the binary side. Whether this is better though is subjective. 🤷